frontend/src/Leagues.tsx

```diff
@@ -0,0 +1,201 @@
+import React from "react";
+import Box from "@mui/joy/Box";
+import { extendedLeagues, leagueTiers, START_YEAR } from "./constants";
+import {
+    HEADER_HEIGHT_PX,
+    HEADER_MARGIN_PX,
+    TABLE_BTN_HEIGHT_PX,
+    TABLE_ELEMENTS_GAP,
+    BUTTON_SELECTOR_HEIGHT,
+} from "./styles/sizes";
+import { LeagueParams, LeagueStats, LeagueTier } from "./types";
+import { getLeagueLabel, range, toPercent } from "./utils";
+import ButtonSelector from "./ButtonSelector";
+import Table, { ColHeaderData, RowData } from "./Table";
+import TableLayout from "./TableLayout";
+
+const TABLE_TOP_POSITION =
+    HEADER_HEIGHT_PX +
+    HEADER_MARGIN_PX +
+    BUTTON_SELECTOR_HEIGHT * 2 +
+    TABLE_BTN_HEIGHT_PX +
+    TABLE_ELEMENTS_GAP * 4;
+
+interface Props {
+    params: LeagueParams;
+    setParams: React.Dispatch<React.SetStateAction<LeagueParams>>;
+    stats: LeagueStats;
+    loading: boolean;
+    error: string | null;
+    refresh: () => void;
+}
+
+export default function Leagues({
+    params,
+    setParams,
+    stats,
+    loading,
+    error,
+    refresh,
+}: Props) {
+    const availableYears = range(START_YEAR, new Date().getFullYear() + 1);
```

Comment:
For MVP I think we should only support 2025. We can support earlier years post-MVP once we figure out a data migrating solution.

Reply:
Plus it's possible nobody cares about earlier years.

Reply:
Limited it to 2025! I left the selector there, even though it has only one option, so that it will still automatically change to the new year in 2026. I'm not sure if I like how it looks with only one option, but can revisit!